lib/src/material_controls.dart

```diff
@@ -68,47 +57,41 @@ class _MaterialControlsState extends State<MaterialControls> {
   }
 
   void _dispose() {
-    widget.controller.removeListener(_updateState);
+    controller.removeListener(_updateState);
     _hideTimer?.cancel();
     _showTimer?.cancel();
     _showAfterExpandCollapseTimer?.cancel();
   }
 
   @override
-  void initState() {
+  void didChangeDependencies() {
+    chewieController = ChewieControllerProvider.of(context);
+    controller = chewieController.videoPlayerController;
+
+    _dispose();
```

Comment:
Unlike `initState`, `didChangeDependencies` can run multiple times throughout the lifecycle of the Widget. Therefore, we may want to add some logic around this `_dispose` / `initialize` code: Only run that dispose and init IF the `controller` has changed.

Reply:
Right, it was also reinitializing when the theme changed. fc1bc16 fixes that.